dvc/scm/git/__init__.py

```diff
@@ -162,6 +163,13 @@ def _add_entry_to_gitignore(entry, gitignore):
                 prefix = "" if last == "\n" else "\n"
             fobj.write("{}{}\n".format(prefix, entry))
 
+    @staticmethod
+    def _escape(entry):
+        # Reference: https://git-scm.com/docs/gitignore#_pattern_format
+        meta_characters = r"[]!*#?"
+
+        return "".join("\\" + x if x in meta_characters else x for x in entry)
```

Comment:
This could be implemented with regexp quite easily and will handle # and ! at the start too. It's up to you though.

Reply:
@Suor , could you give an example of a quite easy one?

Regexp are hard to work with, I was thinking about doing `'[ + re.escape('[]!*#?') + ']' and use that pattern on `sub` but it turns out I need some grouping to use back references. For me it is simpler and easier to use a simple `x in meta_characters`, maybe, instead of a string, we could use a set for some "speed" improvement.